diff --git a/layers/+lang/php/packages.el b/layers/+lang/php/packages.el
index 7cf0bfb..94b90c8 100644
--- a/layers/+lang/php/packages.el
+++ b/layers/+lang/php/packages.el
@@ -15,6 +15,7 @@
         drupal-mode
         eldoc
         flycheck
+        geben
         ggtags
         helm-gtags
         php-auto-yasnippets
@@ -22,6 +23,7 @@
         php-mode
         phpcbf
         phpunit
+        popwin
         ))
 
 (defun php/post-init-company ()
@@ -37,9 +39,23 @@
 (defun php/post-init-flycheck ()
   (spacemacs/add-flycheck-hook 'php-mode))
 
+(defun php/init-geben ()
+  (use-package geben
+    :defer t))
+
 (defun php/post-init-ggtags ()
   (add-hook 'php-mode-local-vars-hook #'spacemacs/ggtags-mode-enable))
 
+(defun php/pre-init-popwin ()
+  (spacemacs|use-package-add-hook popwin
+    :post-config
+    (push
+     '(".*backtrace\*" :regexp t :position bottom)
+     popwin:special-display-config)
+    (push
+     '(".*context\*" :regexp t :position left :stick t :noselect t :width 80)
+     popwin:special-display-config)))
+
 (defun php/post-init-helm-gtags ()
   (spacemacs/helm-gtags-define-keys-for-mode 'php-mode))
 
